src/components/spreadsheet/network-modification-tree-pane-panels-handlers.tsx

```diff
@@ -0,0 +1,103 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { ImperativePanelGroupHandle } from 'react-resizable-panels';
+
+const pixelsToPercents = (elementPixelSize: number, containerPixelSize: number) => {
+    return containerPixelSize ? (elementPixelSize / containerPixelSize) * 100 : 50;
+};
+
+const percentsToPixels = (elementPercentSize: number, containerPixelSize: number) => {
+    return (elementPercentSize * containerPixelSize) / 100;
+};
+
+interface UsePanelsSize {
+    containerRef: React.RefObject<HTMLDivElement>;
+    panelGroupRef: React.RefObject<ImperativePanelGroupHandle>;
+    showRightPanel: boolean;
+    rightComponentDefaultSizePixel: number;
+    rightComponentMinSizePixel: number;
+}
+
+export const usePanelsSize = (props: UsePanelsSize) => {
+    const { containerRef, panelGroupRef, showRightPanel, rightComponentDefaultSizePixel, rightComponentMinSizePixel } =
+        props;
+    const rightComponentMinSizePercentage = containerRef.current?.offsetWidth
+        ? pixelsToPercents(rightComponentMinSizePixel, containerRef.current.offsetWidth)
+        : 20;
+
+    // const rightComponentDefaultSizePixel = 320 + rootNetworks.length * 80;
+    const rightComponentDefaultSizePercentage = containerRef.current?.offsetWidth
+        ? pixelsToPercents(rightComponentDefaultSizePixel, containerRef.current.offsetWidth)
+        : 50;
+
+    const [hasUserInterracted, setHasUserInterracted] = useState(false);
+    const [rightPanelPixelSize, setRightPanelPixelSize] = useState(rightComponentDefaultSizePixel);
+
+    const rightPanelPercentSize = useMemo(
+        () =>
+            containerRef?.current?.offsetWidth
+                ? pixelsToPercents(rightPanelPixelSize, containerRef?.current?.offsetWidth)
+                : 50,
```

Comment:
Shouldn't it use rightComponentDefaultSizePercentage instead of 50 ?

Reply:
I agree, fixed